master/internal/rm/agentrm/agent_resource_manager.go

```diff
@@ -389,12 +389,13 @@ func (*ResourceManager) IsReattachableOnlyAfterStarted() bool {
 
 // MoveJob implements rm.ResourceManager.
 func (a *ResourceManager) MoveJob(msg sproto.MoveJob) error {
-	for _, pool := range a.pools {
-		err := pool.MoveJob(msg)
-		if err != nil {
-			a.syslog.WithError(err).Errorf("failed to move job %s", msg.ID)
-			return err
-		}
+	pool, err := a.poolByName(msg.ResourcePool)
```

Comment:
can you add an error into `poolByName` that is like
```
if pool == "" {
    return nil, errors.New("invalid call: cannot get a resource pool with no name")
}
```
so it is really obvious if we ever miss one